### PR TITLE
gate snippet folders and their permissions behind `snippet_collections` premium feature

### PR DESCRIPTION
**Pull request by EmmadUsmani**
Closes https://github.com/metabase/metabase-private/issues/76

### Description

Snippet folders and their permissions were not probably gated behind premium feature flags. This PR fixes that according to [our plan](https://www.notion.so/metabase/July-23-Unraveling-of-Feature-Flags-6300baadb8f54fadace6cf0e2fc11a2b).

### How to verify

1. In EE with a token that has the `snippet_collections` feature, Create a SQL question
2. Open the SQL snippets sidebar and create one.
3. Create a snippet folder and move the snippet inside of there.
4. Confirm that you can change permissions at the root level and the folder.
5. Go to admin settings and remove the token.
6. Go back to the snippets sidebar, the snippet should be displayed without any folders, and the permissions options should not be visible.
7. Same should be true on OSS.

### Demo

https://github.com/metabase/metabase/assets/37751258/c82e4e9f-89a0-4e71-bfb6-14f575ef8d2d

EE with token

https://github.com/metabase/metabase/assets/37751258/69a8fe92-d48f-4dd1-9051-4a19fc0e8765

EE without token / OSS

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: e2e/test/scenarios/native/snippets/snippet-permissions.cy.spec.js [hunk 1]
```diff
@@ -55,8 +101,6 @@ describeEE("scenarios > question > snippets", () => {
       collection_id: null,
     });
 
-    cy.intercept("GET", "api/collection/*").as("collection");
```
Comment (EmmadUsmani) [inline e2e/test/scenarios/native/snippets/snippet-permissions.cy.spec.js:58]: The alias was never actually used

File: enterprise/frontend/src/metabase-enterprise/snippets/components/CollectionOptionsButton.jsx [hunk 2]
```diff
@@ -74,7 +71,12 @@ export default class CollectionOptionsButton extends Component {
         onClick: () => onEdit(collection),
       });
     }
-    if (user && user.is_superuser) {
+    if (
+      user &&
+      user.is_superuser &&
+      (hasPremiumFeature("content_management") ||
```
Comment (ranquild) [inline enterprise/frontend/src/metabase-enterprise/snippets/components/CollectionOptionsButton.jsx:77]: I'd like to keep `hasPremiumFeature` checks only to plugin initialization code. Can we move it there?

**Review by ranquild — COMMENTED** [review]
Comments inline

File: enterprise/frontend/src/metabase-enterprise/snippets/components/CollectionOptionsButton.jsx (hunk 2 above)
Comment (EmmadUsmani) [reply enterprise/frontend/src/metabase-enterprise/snippets/components/CollectionOptionsButton.jsx:77]: Yup I can do that along with changing the flag to `snippet_collections`

Comment (EmmadUsmani) [reply enterprise/frontend/src/metabase-enterprise/snippets/components/CollectionOptionsButton.jsx]: This check ended up being redundant (component is already gated by a plugin) so I removed it.

**Review by alxnddr — APPROVED** [review]
Looks great 👍

**Review by ranquild — APPROVED** [review]

**Pull request merged**